### PR TITLE
A FakeSimpleNagios class to avoid side effects

### PR DESCRIPTION
**Pull request by itkovian**

### CODE REVIEW

File: lib/vsc/utils/nagios.py [hunk 1]
```diff
@@ -408,6 +409,43 @@ def __str__(self):
         return "%s | %s" % (self.message, ' '.join(perf))
 
 
+class FakeSimpleNagios(NagiosResult):
+    """Absolutely no side effects.
+
+    For use in long running services that have no access to the locations where
```
Comment (stdweird) [inline lib/vsc/utils/nagios.py:415]: such as?

File: lib/vsc/utils/nagios.py [hunk 2]
```diff
@@ -408,6 +409,43 @@ def __str__(self):
...
+    Nagios results would be stored.
+    """
+
+    def __init__(self, **kwargs):
+        logging.debug("empty __init__")
```
Comment (stdweird) [inline lib/vsc/utils/nagios.py:420]: why not add the args  to the message? now this is totally useless.

File: lib/vsc/utils/script_tools.py [hunk 3]
```diff
@@ -70,6 +71,7 @@ def _script_name(full_name):
 
 DEFAULT_OPTIONS = {
     'disable-locking': ('do NOT protect this script by a file-based lock', None, 'store_true', False),
+    'disable-nagios': ('do NOT generate a nagios cache file. Implies no locking.', None, 'store_true', False),
```
Comment (stdweird) [inline lib/vsc/utils/script_tools.py:74]: why does it imly no locking. that sounds pretty dangerous

File: lib/vsc/utils/nagios.py (hunk 1 above)
Comment (itkovian) [reply lib/vsc/utils/nagios.py:415]: xdmod-consumer

File: lib/vsc/utils/script_tools.py (hunk 3 above)
Comment (itkovian) [reply lib/vsc/utils/script_tools.py:74]: yeah, that's wrong, need to fix the comment.

File: lib/vsc/utils/nagios.py (hunk 1 above)
Comment (stdweird) [reply lib/vsc/utils/nagios.py:415]: and why wouldn;t it have access to the nagios locations?

Comment (itkovian) [reply lib/vsc/utils/nagios.py:415]: because it is running as the XDMoD user

Comment (stdweird) [reply lib/vsc/utils/nagios.py:415]: hmmmm
we had stuff in the past like this. i think we settled on "running as root is better than running without monitoring".
i guess we can make it write into paths it owns, we can always see how to have nrpe actually read it out

**Pull request closed without merging**